### PR TITLE
CLI for merging rewrite rules.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

**Comment by virgil-serbanuta** [comment]
@ehildenb Could you take a look and see if things make sense? I mostly worry about this result:
https://github.com/kframework/kore/pull/983/files#diff-8673467e4d017e50c56dcc0de3358bd4R1
since it wasn't what I expected intuitively.

File: docs/2019-09-09-Combining-Rewrite-Axioms.md [hunk 1]
```diff
@@ -180,6 +180,47 @@ By applying this iteratively, we get
 (⌈β₁(X₁)∧α₂(X₂)⌉ ∧ ⌈β₂(X₂)∧α₃(X₃)⌉ ∧ ... ⌈βₙ₋₁(Xₙ₋₁)∧αₙ(Xₙ)⌉ ∧ α₁(X₁))→••…•βₙ(Xₙ)
 ```
 
+### Removing Substitutions
```
Comment (traiansf) [inline docs/2019-09-09-Combining-Rewrite-Axioms.md]: Maybe add an example here.

File: kore/src/Kore/Step/Rule.hs [hunk 2]
```diff
@@ -66,13 +67,53 @@ import Kore.Attribute.Priority
 import Kore.Debug
 import Kore.Error
 import Kore.IndexedModule.IndexedModule
+import Kore.Internal.Alias
+    ( Alias (..)
+    )
 import Kore.Internal.ApplicationSorts
-import Kore.Internal.TermLike as TermLike
+import Kore.Internal.TermLike
```
Comment (traiansf) [inline kore/src/Kore/Step/Rule.hs]: Why do you need all these imports? Why weren't they needed before? It seems your change does not concern them.

File: kore/src/Kore/Step/Rule/Combine.hs [hunk 3]
```diff
@@ -118,3 +170,54 @@ renameRuleVariable
...
+            , right = finalRight
+            , ensures = finalEnsures
+            , attributes = def
+            }
+
+        substitutedVariables = Substitution.variables substitution
+
+    Monad.unless (finalRule `Rule.isFreeOf` substitutedVariables)
+        (error
+            (  "Substituted variables not remove from the rule, cannot throw "
```
Comment (traiansf) [inline kore/src/Kore/Step/Rule/Combine.hs]:
```suggestion
            (  "Substituted variables not removed from the rule, cannot be discarded "
```

File: docs/2019-09-09-Combining-Rewrite-Axioms.md (hunk 1 above)
Comment (virgil-serbanuta) [reply docs/2019-09-09-Combining-Rewrite-Axioms.md]: Done.

**Pull request merged**